torch/csrc/jit/script/init.cpp

```diff
@@ -90,43 +90,30 @@ struct VISIBILITY_HIDDEN PythonValue : public SugaredValue {
         throw ErrorReport(loc) << "type mismatch at argument " << i << ": expected "
                                << arguments[i]->str() << ", but got " << inputs[i]->type()->str();
     }
-    // We have to do this check here, because implementation of this function is tightly
-    // coupled with the impl for PythonOp in the interpreter. Right now it assumes that
-    // all inputs taken from the stack are Tensors, so that's what we have to do.
-    ensureTensors(loc, inputs);
-
     if (attributes.size() > 0)
       throw ErrorReport(loc) << "keyword arguments in Python calls aren't supported";
-    Graph& g = *m.graph();
-
-    // this python object might be a @trace or @script function/module
-    // if so, inline the graph rather than calling the python
-
-    if(py::isinstance<Module>(self)) {
-      Module& mod = py::cast<Module&>(self);
-      if (Method * forward = mod.find_method("forward")) {
-        // This code path should only get called for Modules that are really
-        // wrappers around pure script/traced functions. Modules with parameters
-        // should be submodules of the caller, and thus will be represented as
-        // ModuleValue and not go through here.
-        if (mod.get_parameters().size() != 0) {
-          throw ErrorReport(loc) << "Attempted to inline a Module with parameters. "
-            "Stateful modules to be inlined must be submodules of the callee.";
-        }
-        std::vector<torch::jit::NamedValue> named_inputs;
-        for (auto inp : inputs)
-          named_inputs.push_back(NamedValue(loc, "", inp));
-        return packOutputs(*m.graph(), m.emit_call_to(loc, *forward, named_inputs, {}));
-      }
+    std::vector<Argument> arguments_args, returns_args;
```

Comment:
This is redundant. You can change `getArgumentTypes` to `getSchema` and have it return a `FunctionSchema`. You can then call tryMatchSchema. All of the error checking about argument sizes kwargs are redundant.